From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sotr <i@omc.hk>
Date: Wed, 15 Apr 2020 02:49:56 +0700
Subject: [PATCH] Don't run entity collision code if not needed

Will not run if max entity craming is disabled and
the max collisions per entity is less than or equal to 0

This commit was basically referenced on Tuinity#7131da4.

diff --git a/src/main/java/net/minecraft/server/EntityLiving.java b/src/main/java/net/minecraft/server/EntityLiving.java
index 284826897995c278c0b76a4114fad4e60309a1f9..c1c7e7dca3599353793bbfa4635f93554d9c737c 100644
--- a/src/main/java/net/minecraft/server/EntityLiving.java
+++ b/src/main/java/net/minecraft/server/EntityLiving.java
@@ -2663,10 +2663,16 @@ public abstract class EntityLiving extends Entity {
     protected void doTick() {}
 
     protected void collideNearby() {
+        // Akarin start - don't run getEntities if we're not going to use its result
+        if (world.paperConfig.maxCollisionsPerEntity <= 0) return;
+
+        int i = this.world.getGameRules().getInt(GameRules.MAX_ENTITY_CRAMMING);
+        if (i <= 0) return;
+        // Akarin end
         List<Entity> list = this.world.getEntities(this, this.getBoundingBox(), IEntitySelector.a(this));
 
         if (!list.isEmpty()) {
-            int i = this.world.getGameRules().getInt(GameRules.MAX_ENTITY_CRAMMING);
+            // int i = this.world.getGameRules().getInt(GameRules.MAX_ENTITY_CRAMMING); // Akarin - moved up
             int j;
 
             if (i > 0 && list.size() > i - 1 && this.random.nextInt(4) == 0) {
